### PR TITLE
Fix/sparsity pattern gating

### PR DESCRIPTION
**Pull request by mohitpundir**
This PR adds handlers for missing primitives for JAX linear algebra operations at the element level, such as `jnp.linalg.inv`, `jnp.linalg.eig` and also for `jnp.flip` and `jnp.stack`.  In this PR, we fix the edge cases with `dot_general`  such as:

-  over-conservatism if one of the operands is just linearly dependent on the solution. Such a case was never checked for how the operand is dependent on the solution because only a nonlinear dependency will lead to a non-zero Hessian entry. The fix records an operand's self-second-order coupling only if it is nonlinear in the solution; the cross second-order coupling between operands is always kept. 
-  when we `dot_general` contract a value that depends on the solution with a constant for example `stress @ n` (per segment stress times a fixed normal) along an interface, the tracer switched to fallback option as it didnot understand which axes the primitive `dot_general` was contracting over and fallback on `total_union` and hence assumed the results depends on every degree of freedom which made the sparsity pattern dense block. The fix tells `dot_general` which axis is being summed over, so dependencies are combined only along the axis and thus the pattern stays block-diagonal, and no real entry is lost.

This PR also adds a warning for the primitives that are not considered. 

**Checklist:**
- [x] I have read the contributing guidelines.
- [x] My code follows the style guidelines of this project. `tatva` uses `ruff`.
- [x] I have performed a self-review of my own code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Review by zrlf — APPROVED** [review]
awesome🤩🙌
